src/components/numeral-date/numeral-date.test.tsx

```diff
@@ -0,0 +1,1995 @@
+    expect(
+      screen.getByText("Day should be a number within a 1-31 range.")
+    ).toBeVisible();
+  });
+
+  it("should render the validation message when the user types a value in the `Day` field less than 1 and tabs", async () => {
+    const user = userEvent.setup({ advanceTimers: jest.advanceTimersByTime });
+    render(
+      <CarbonProvider validationRedesignOptIn>
+        <NumeralDate
+          value={{ dd: "", mm: "", yyyy: "" }}
+          onChange={() => {}}
+          enableInternalError
+        />
+      </CarbonProvider>
+    );
+
+    const dayInput = screen.getByRole("textbox", { name: "Day" });
+    const monthInput = screen.getByRole("textbox", { name: "Month" });
+    const yearInput = screen.getByRole("textbox", { name: "Year" });
+    await user.type(dayInput, "0");
+    await user.tab();
+
+    expect(dayInput).toBeInvalid();
+    expect(monthInput).toBeInvalid();
+    expect(yearInput).toBeInvalid();
+    expect(
+      screen.getByText("Day should be a number within a 1-31 range.")
+    ).toBeVisible();
+  });
+
+  it("should render the validation message when the user types `02` in the `Month` field and tabs when the `Day` field has a value greater than 28", async () => {
+    const user = userEvent.setup({ advanceTimers: jest.advanceTimersByTime });
+    render(
+      <CarbonProvider validationRedesignOptIn>
+        <NumeralDate
+          value={{ dd: "29", mm: "", yyyy: "2021" }}
+          onChange={() => {}}
+          enableInternalError
+        />
+      </CarbonProvider>
+    );
+
+    const monthInput = screen.getByRole("textbox", { name: "Month" });
+    const dayInput = screen.getByRole("textbox", { name: "Day" });
+    const yearInput = screen.getByRole("textbox", { name: "Year" });
+    await user.type(monthInput, "02");
+    await user.tab();
+
+    expect(dayInput).toBeInvalid();
+    expect(monthInput).toBeInvalid();
+    expect(yearInput).toBeInvalid();
+    expect(
+      screen.getByText("Day in February should be a number within 1-28.")
+    ).toBeVisible();
+  });
+
+  it("should not render the validation message when the user types '02' in the`Month` field and tabs when the `Day` field has a valid value", async () => {
+    const user = userEvent.setup({ advanceTimers: jest.advanceTimersByTime });
+    render(
+      <CarbonProvider validationRedesignOptIn>
+        <NumeralDate
+          value={{ dd: "29", mm: "", yyyy: "2020" }}
+          onChange={() => {}}
+          enableInternalError
+        />
+      </CarbonProvider>
+    );
+
+    const monthInput = screen.getByRole("textbox", { name: "Month" });
+    await user.type(monthInput, "02");
+    await user.tab();
+
+    expect(
+      screen.queryByText("Day in February should be a number within 1-28.")
+    ).not.toBeInTheDocument();
+  });
+
+  it("should use the current year and not render the validation message when the user types '02' in the`Month` field and tabs when the `Day` field has a valid value", async () => {
```

Comment:
**comment** (non-blocking): I'm not sure what "use the current year" has to do with this test to be honest. It's testing February 28th which of course is a valid date in any year. So I would change the description to just be something like "valid dates in February should not render the validation message", although arguably that's not even needed in addition to the other tests we have.

If we do want to test the functionality of taking the current date, we'd have to mock native date functions (`Date.prototype.getFullYear` I think, having looked at the component code) - and well I'll leave it up to you to judge if that's worth it 😆

Reply:
This is mainly for coverage, it's intentionally testing that when the year field is empty it uses the current year to validate, we don't need to mock it as it's always valid.

Reply:
I agree there's no need to mock anything for a test on February 28th as that is valid in any year, but:
- there's likely no point in a test for "february 28th is a valid date, at least if you leave out the year"
- even if we decide there is, the description of the test doesn't match what it's actually doing

I get that we need to do a case where there's no year given to hit coverage, but I think there is actually value in testing that - and I think the best way to do that is to test with February 29th and mock `getFullYear` to return say 2024 in one test (where it should be valid) and say 2023 in another (where it shouldn't be). If you disagree that that's worth a proper test, then fair enough, but at least then let's put in a "dummy test" with a comment saying it's only for coverage, not have a description that looks like it's testing the "use current year" functionality but isn't really (except in the coverage sense).

Reply:
I'm not sure I fully agree, we have tests for leap years, this test is purely to assert that it gets the full year for the current date so the fact it's not testing leap years isn't critical for me. I'd prefer to do this than have to mock the required functionality so as not to have failures when that updates every year etc